events/static/events/js/script.js

```diff
@@ -0,0 +1,71 @@
+async function initMap() {
+    map = new google.maps.Map(document.getElementById("map"), {
+      center: { lat: 40.71159438471519, lng: -73.97424027294727 },
+      zoom: 12,
+      mapId: '9a7e26117f10196d'
+    });
+        const responseEvent = await fetch('get_data/');
+        const responseLocation = await fetch('get_locations/');
+        const event_data = await responseEvent.json();
+        const location_objects = await responseLocation.json();
+        const location_data = event_data.location_data;
+        const locations = location_objects.locations;
+        const locationmap = new Map();
+        for(let i =0; i<locations.length;i++){
+            locationmap.set(locations[i].pk,[locations[i].fields.latitude,locations[i].fields.longitude])
+        }
+        let duplicateLocationsMap = {};
+        for(let i=0; i<location_data.length;i++){
+            let loc = location_data[i].fields;
+            if(loc.is_active==true){
+                if(!duplicateLocationsMap[loc.event_location]){
+                    duplicateLocationsMap[loc.event_location] = [];
+                }
+                duplicateLocationsMap[loc.event_location].push({name:loc.event_name,id:event_data.location_data[i].pk});
+            }
+        }
+        for(let i =0; i<location_data.length;i++){
+            
+            if(location_data[i].fields.is_active == true){
+                let lat = -1, lng = -1;
+                if(locationmap.get(location_data[i].fields.event_location)!=null){
+                    lat = locationmap.get(location_data[i].fields.event_location)[0];
+                    lng = locationmap.get(location_data[i].fields.event_location)[1];
+                }
+            const marker = new google.maps.Marker({
+                position: {lat: lat, lng:lng},
+                map,
+                title: location_data[i].fields.event_name,
+                icon: {
+                    url: "https://www.svgviewer.dev/static-svgs/490211/marker.svg",
```

Comment:
can we download this svg and keep it in our repo? Reason being later on if svgviewer changes anything on this link we are directly dependent

Reply:
Yup I'll add it to the repo

Reply:
I checked if this could be done, but the svg has to be hosted on the web. Either that or we have to retrieve it using an AJAX request to fetch the file and convert it into a data URI. I don't think that would be necessary for our project.